### PR TITLE
[#536] Fix ex.Utils.clamp min/max handling

### LINKED ISSUE #536: Update ex.Util.clamp to be smarter
Here is the code :P

``` typescript
function clamp(val, min, max) {
   return Math.min(Math.max(min, val), max);
}
```

### PR DESCRIPTION
**Pull request by FerociousQuasar**
Closes #536 

## Changes:
- Update ex.Util.clamp to use Math libraries Min and Max
- Add unit tests for ex.Util.clamp

### CODE REVIEW

**Comment by kamranayub** [comment]
Also, I'm thinking to be good citizens we should ensure this function has proper tests around it.

**Comment by FerociousQuasar** [comment]
Roger.
Will add unit tests to make sure this behaves as expected.
Will update CHANGELOG.md with the changes.
Will update commit message to meet the appropriate standard.

**Comment by eonarheim** [comment]
@jedeen @FerociousQuasar Let's not worry about the commit messages on this one since we'll squash the messages.

Any issues @excaliburjs/core-contributors? LGTM :shipit:

**Comment by kamranayub** [comment]
Only two thoughts, could you add a couple tests that also test boundary numbers (0, Infinity, -Infinity) to ensure it still works as expected?

File: src/spec/UtilSpec.ts [hunk 1]
```diff
@@ -16,4 +16,10 @@ describe('Utility functions', () => {
       expect(ex.Util.getSideFromVector(ex.Vector.Up)).toBe(ex.Side.Top);
       expect(ex.Util.getSideFromVector(ex.Vector.Down)).toBe(ex.Side.Bottom);
    });
+
+   it('can clamp a number to a maximum and minimum', () => {
+      expect(ex.Util.clamp(0, 10, 20)).toBe(10);
+      expect(ex.Util.clamp(15, 10, 20)).toBe(15);
+      expect(ex.Util.clamp(30, 10, 20)).toBe(20);
```
Comment (kamranayub) [inline src/spec/UtilSpec.ts]: Could you add another couple tests for Infinity and -Infinity to test those boundaries?

File: src/engine/Util/Util.ts [hunk 2]
```diff
@@ -104,7 +104,7 @@ module ex.Util {
    }
 
    export function clamp(val, min, max) {
```
Comment (kamranayub) [inline src/engine/Util/Util.ts]: Could you add the `number` types to each parameter to make sure people don't accidentally pass in other types?
